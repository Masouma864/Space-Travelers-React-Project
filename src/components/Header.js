import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../images/planet.png';

const Header = () => (
    <div className='header-container'>
        <div className="left-header">
            <img src={planet} alt="planetImage" width="60px" />
            <h3>Space Travelers Hub</h3>
        </div
        <ul className="flex">
      <li>
        <NavLink
          to="/"
          style={(isActive) => ({
            textDecoration: isActive ? 'underline' : 'none',
          })}
        >
          Rockets
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/missionspage"
          style={(isActive) => ({
            textDecoration: isActive ? 'underline' : 'none',
          })}
        >
          Missions
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/myprofilepage"
          style={(isActive) => ({
            textDecoration: isActive ? 'underline' : 'none',
          })}
        >
          My Profile
        </NavLink>
      </li>
    </ul>
  </div>

);