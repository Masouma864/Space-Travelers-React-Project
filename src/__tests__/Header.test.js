import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../components/Header';

describe('Header', () => {
  it('should render header correctly', () => {
    render(
      <Router>
        <Header />
      </Router>,
    );
    const rocketsTag = screen.getByText(/rockets/i);

    expect(rocketsTag).toMatchSnapshot();
  });
  it('should render header correctly', () => {
    render(
      <Router>
        <Header />
      </Router>,
    );
    const missionsTag = screen.getByRole('link', { name: 'Missions' });

    expect(missionsTag).toMatchSnapshot();
  });
});
