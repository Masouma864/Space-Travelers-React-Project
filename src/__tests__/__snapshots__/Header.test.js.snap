// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render header correctly 1`] = `
<a
  aria-current="page"
  class="rockets active"
  href="/"
  style="text-decoration: underline;"
>
  Rockets
</a>
`;

exports[`Header should render header correctly 2`] = `
<a
  class="missions"
  href="/missionspage"
>
  Missions
</a>
`;
